src/test/scala/chiselTests/BoringUtilsTapSpec.scala

```diff
@@ -109,6 +145,63 @@ class BoringUtilsTapSpec extends ChiselFlatSpec with ChiselRunners with Utils wi
     )()
   }
 
+  it should "work upwards from grandchild to grandparent through when" in {
+    class Bar(grandParentData: Data) extends RawModule {
+      val out = IO(Bool())
+      out := BoringUtils.tapAndRead(grandParentData)
+    }
+    class Foo(parentData: Data) extends RawModule {
+      when(true.B) {
+        val bar = Module(new Bar(parentData))
+      }
+    }
+    class Top extends RawModule {
+      val parentWire = Wire(Bool())
+      parentWire := DontCare
+      val foo = Module(new Foo(parentWire))
+    }
+    val chirrtl = circt.stage.ChiselStage.emitCHIRRTL(new Top)
+
+    // The connect should be at the end of the when block.
+    matchesAndOmits(chirrtl)(
+      "    when UInt<1>(0h1) :",
+      "      inst bar of Bar",
+      "      connect bar.out_bore, out_bore"
+    )()
```

Comment:
We really need something better than `matchesAndOmits` (FileCheck) as this test will, I think, pass without this change. Given that, the test might be better written as simply running `emitFIRRTLDialect` which will cause CIRCT to check that this isn't busted. Could you check if that would cause this to fail before the commit and pass after?

Reply:
The whitespace is what ensures this fails without the change, just re-checked that this is the case.
Whoa, TIL about `emitFIRRTLDialect`!